cmd/clusterctl/client/upgrade.go

```diff
@@ -208,9 +208,6 @@ func parseUpgradeItem(ref string, providerType clusterctlv1.ProviderType) (*clus
 			},
 			ProviderName: name,
 			Type:         string(providerType),
-			// The value for the following fields will be retrieved while
-			// creating the custom upgrade plan.
-			WatchedNamespace: "",
```

Comment:
As per https://github.com/kubernetes-sigs/cluster-api/pull/5288/files#r713637308 this behaviour can't be broken unless the api is bumped.

Reply:
+1 Imho we have to keep the field, see https://github.com/kubernetes-sigs/cluster-api/pull/5288#discussion_r713637308